connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/InboundExecutableRegistryImpl.java

```diff
@@ -0,0 +1,204 @@
+package io.camunda.connector.runtime.inbound.executable;
+
+import com.google.common.collect.EvictingQueue;
+import io.camunda.connector.api.inbound.Health;
+import io.camunda.connector.api.inbound.InboundConnectorExecutable;
+import io.camunda.connector.api.inbound.webhook.WebhookConnectorExecutable;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorContextFactory;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorDefinitionImpl;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorFactory;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorReportingContext;
+import io.camunda.connector.runtime.inbound.executable.InboundExecutableEvent.Activated;
+import io.camunda.connector.runtime.inbound.executable.InboundExecutableEvent.Deactivated;
+import io.camunda.connector.runtime.inbound.webhook.WebhookConnectorRegistry;
+import io.camunda.connector.runtime.metrics.ConnectorMetrics.Inbound;
+import io.camunda.zeebe.spring.client.metrics.MetricsRecorder;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+
+public class InboundExecutableRegistryImpl implements InboundExecutableRegistry {
+
+  @Value("${camunda.connector.inbound.log.size:10}")
+  private int inboundLogsSize;
+
+  private final InboundConnectorFactory connectorFactory;
+  private final InboundConnectorContextFactory connectorContextFactory;
+  private final MetricsRecorder metricsRecorder;
+  private final WebhookConnectorRegistry webhookConnectorRegistry;
+
+  private final BlockingQueue<InboundExecutableEvent> eventQueue;
+  private final ExecutorService executorService;
+
+  private final ConcurrentHashMap<UUID, ActiveExecutable> executables = new ConcurrentHashMap<>();
```

Comment:
Do we need a `ConcurrentHashMap` here?

Reply:
I removed concurrent hash maps and replaced them with a single synchronization point. I'd like the event processing in the executable registry to be thread-safe because we might want to scale up the thread pool in the future.